File: leetcode/lc-us/cpp/242.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        }
        int count[26] = {0};
        for (int i = 0; i < s.size(); i++) {
            count[s[i] - 'a']++;
            count[t[i] - 'a']--;
        }
        return all_of(count, count + 26, [](int i) { return i == 0; });
    }
};

File: leetcode/lc-us/rust/242.rs
impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        let mut cnt = vec![0; 26];
        for c in s.bytes() {
            cnt[c as usize - 'a' as usize] += 1;
        }
        for c in t.bytes() {
            cnt[c as usize - 'a' as usize] -= 1;
        }
        for i in 0..26 {
            if cnt[i] != 0 {
                return false;
            }
        }
        true
    }
}